Controllers/DriverController.cs

```diff
@@ -0,0 +1,157 @@
+using Microsoft.AspNetCore.Mvc;
+using advent_appointment_booking.DTOs;
+
+namespace advent_appointment_booking.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class DriverController : ControllerBase
+    {
+        private readonly IDriverService _driverService;
+
+        public DriverController(IDriverService driverService)
+        {
+            _driverService = driverService;
+        }
+
+        // Create a new driver
+        [HttpPost]
+        [Authorize(Policy = Policy.RequireTruckingCompanyRole)]
+        public async Task<IActionResult> CreateDriver([FromBody] Driver driver)
+        {
+            if (!ModelState.IsValid)
+            {
+                var errors = ModelState.Values.SelectMany(v => v.Errors).Select(e => e.ErrorMessage);
+                return BadRequest(new { message = "Validation failed", errors });
+            }
+
+            try
+            {
+                var result = await _driverService.CreateDriver(driver);
+                return Ok(result);
+            }
+            catch (Exception ex)
+            {
+                if (ex.Message.Contains("Trucking company doesn't exist"))
+                {
+                    return NotFound(new { message = ex.Message });
+                }
+                else if (ex.Message.Contains("Driver with the same plate number already exists") ||
+                         ex.Message.Contains("Driver with the same phone number already exists"))
+                {
+                    return Conflict(new { message = ex.Message });
+                }
+
+                return BadRequest(new { message = ex.Message });
+            }
+        }
+
+        // Get details of a driver by ID
+        [HttpGet("{driverId}")]
+        public async Task<IActionResult> GetDriver(int driverId)
+        {
+            var driver = await _driverService.GetDriver(driverId);
+
+            if (driver == null)
+            {
+                return NotFound(new { message = "Driver not found" });
+            }
+
+            // Map the Driver entity to DriverDTO
+            var driverDto = new DriverDTO
+            {
+                DriverId = driver.DriverId,
+                TrCompanyId = driver.TrCompanyId,
+                DriverName = driver.DriverName,
+                PlateNo = driver.PlateNo,
+                PhoneNumber = driver.PhoneNumber,
+                TruckingCompany = new TruckingCompanyDTO
+                {
+                    TrCompanyId = driver.TruckingCompany.TrCompanyId,
+                    TrCompanyName = driver.TruckingCompany.TrCompanyName,
+                    GstNo = driver.TruckingCompany.GstNo,
+                    TransportLicNo = driver.TruckingCompany.TransportLicNo,
+                    Email = driver.TruckingCompany.Email,
+                    CreatedAt = driver.TruckingCompany.CreatedAt,
+                    UpdatedAt = driver.TruckingCompany.UpdatedAt
+                }
+            };
```

Comment:
this we can write inside dedicated Service file to keep business logic separate from controller